### PR TITLE
[FLINK-32585] Filter javax.xml.bind:jaxb-api false positive for Pulsar connector

### PR DESCRIPTION
**Pull request by tisonkun**
## What is the purpose of the change

Add one more filter to suppress false positive.

## Brief change log

Trivial.

## Verifying this change

Semantically. I don't find a way to run the check locally properly..

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: tools/ci/flink-ci-tools/src/main/java/org/apache/flink/tools/ci/licensecheck/JarFileChecker.java [hunk 1]
```diff
@@ -216,6 +216,10 @@ private static int findNonBinaryFilesContainingText(
                             path ->
                                     !pathStartsWith(
                                             path, "/org/apache/pulsar/shade/org/glassfish/jersey/"))
+                    .filter(
+                            path ->
+                                    !pathStartsWith(
+                                            path, "/org/apache/pulsar/shade/javax/xml/bind/"))
```
Comment (zentol) [inline tools/ci/flink-ci-tools/src/main/java/org/apache/flink/tools/ci/licensecheck/JarFileChecker.java:222]: please move this up next to the other javax exceptions

**Review by zentol — CHANGES_REQUESTED** [review]

**Review by nlu90 — APPROVED** [review]

**Review by zentol — APPROVED** [review]

**Comment by tisonkun** [comment]
I'm going to merge this patch tomorrow and check if the Pulsar connector can pass with this patch.

**Comment by tisonkun** [comment]
Emm..I think I should merge it now so that the SNAPSHOT can be updated.

**Comment by tisonkun** [comment]
Cross ref - https://github.com/apache/flink-connector-pulsar/pull/54

**Pull request merged**